pkg/node/fsm/sync_state.go

```diff
@@ -163,19 +175,38 @@ func (a *SyncState) Block(p peer.Peer, block *proto.Block) (State, Async, error)
 	}
 	metrics.FSMKeyBlockReceived("sync", block, p.Handshake().NodeName)
 	zap.S().Named(logging.FSMNamespace).Debugf("[Sync][%s] Received block %s", p.ID(), block.ID.String())
+
 	internal, err := a.internal.Block(block)
 	if err != nil {
 		return newSyncState(a.baseInfo, a.conf, internal), nil, a.Errorf(err)
 	}
-	return a.applyBlocks(a.baseInfo, a.conf.Now(a.baseInfo.tm), internal)
+	if !a.baseInfo.enableLightMode {
+		return a.applyBlocksWithSnapshots(a.baseInfo, a.conf.Now(a.baseInfo.tm), internal)
+	}
+	return a, nil, nil
```

Comment:
Why is it no-op if node in the light mode?

Reply:
We added block in internal in previous lines and wait snapshots